src/components/app/App.js

```diff
@@ -12,40 +12,22 @@ class App extends Component {
     this.state = {
       movies: [],
       errorCode: null,
-      selectedMovie: { id: '' }
+      selectedMovie: null
     }
   }
 
   componentDidMount = async () => {
-    if (!this.state.movies.length) {
-      this.fetch()
-    }
+    this.fetch('movies')
   }
   
-  fetch = async () => {
-    try {
-      const fetchedMovies = await getApiData('movies');
-      this.setState({movies: fetchedMovies.movies});
-    } catch (e) {
-      this.setState({errorCode: e.message});
-    }
-  }
-
-  selectMovie = async (id) => {
+  fetch = async (endPoint) => {
     try {
-      let movieIDs
-      if (this.state.movies.length) {
-        movieIDs = this.state.movies.map(movie => movie.id.toString())
+      if (endPoint.includes('/')) {
+        let fetchedMovie = await getApiData(endPoint)
+        this.setState({selectedMovie: fetchedMovie.movie})
       } else {
-        const fetchedMovies = await getApiData('movies');
-        this.setState({movies: fetchedMovies.movies});
-        movieIDs = fetchedMovies.movies.map(movie => movie.id.toString())
-      }
-      if (movieIDs.includes(id)) {
-        const match = await getApiData(`movies/${id}`);
-        this.setState({selectedMovie: match.movie});
-      } else {
-        throw new Error('404');
+        let fetchedMovies = await getApiData(endPoint)
```

Comment:
Didn't even think about the error handling in the API fetch, this is an awesome call